aquadoggo/src/materializer/tasks/blob.rs

```diff
@@ -0,0 +1,274 @@
+                .await
+                .map_err(|err| TaskError::Failure(err.to_string()))?
+                .unwrap();
+            Ok(vec![document])
+        }
+
+        // This task is about an updated blob piece document that may be used in one or more blob documents.
+        SchemaId::BlobPiece(_) => get_related_blobs(&input_view_id, &context).await,
+        _ => Err(TaskError::Critical(format!(
+            "Unknown system schema id: {}",
+            schema
+        ))),
+    }?;
+
+    // The related blobs are not known yet to this node so we mark this task failed.
+    if updated_blobs.is_empty() {
+        return Err(TaskError::Failure(
+            "Related blob does not exist (yet)".into(),
+        ));
+    }
+
+    // Materialize all updated blobs to the filesystem.
+    for blob_document in updated_blobs.iter() {
+        // Get the raw blob data.
+        let blob_data = context
+            .store
+            .get_blob_by_view_id(blob_document.view_id())
+            .await
+            // We don't raise a critical error here, as it is possible that this method returns an
+            // error.
+            .map_err(|err| TaskError::Failure(err.to_string()))?
+            .unwrap();
+
+        // Compose, and when needed create, the path for the blob file.
+        let base_path = match &context.config.base_path {
+            Some(base_path) => base_path,
+            None => return Err(TaskError::Critical("No base path configured".to_string())),
+        };
+
+        let blob_dir = base_path
+            .join(BLOBS_DIR_NAME)
+            .join(blob_document.id().as_str());
+
+        fs::create_dir_all(&blob_dir).map_err(|err| TaskError::Critical(err.to_string()))?;
+        let blob_view_path = blob_dir.join(blob_document.view_id().to_string());
+
+        // Write the blob to the filesystem.
+        info!("Creating blob at path {blob_view_path:?}");
+
+        let mut file = File::create(&blob_view_path).unwrap();
+        file.write_all(blob_data.as_bytes()).unwrap();
+
+        // create a symlink from `../documents/<document_id>` -> `../<document_id>/<current_view_id>`
+        if is_current_view(&context.store, blob_document.view_id()).await? {
+            info!("Creating symlink from document id to current view");
+
+            let link_path = base_path
+                .join(BLOBS_DIR_NAME)
+                .join(BLOBS_SYMLINK_DIR_NAME)
+                .join(blob_document.id().as_str());
+
+            let _ = fs::remove_file(&link_path);
+
+            symlink(blob_view_path, link_path)
+                .map_err(|err| TaskError::Critical(err.to_string()))?;
+        }
+    }
+
+    Ok(None)
+}
+
+/// Retrieve blobs that use the targeted blob piece as one of their fields.
+async fn get_related_blobs(
+    target_blob_piece: &DocumentViewId,
+    context: &Context,
+) -> Result<Vec<StorageDocument>, TaskError> {
+    // Retrieve all blob documents from the store
```

Comment:
Use custom SQL query here.

Reply:
There might be some re-use possibilities once both open blob PRs are merged so I will wait on refactoring this.